client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/GenericKaaClient.java

```diff
@@ -497,4 +500,88 @@
      * @return client's private key
      */
     PrivateKey getClientPrivateKey();
+
+    /**
+     * Generate new access token for a current endpoint
+     */
+    String refreshEndpointAccessToken();
+
+    /**
+     * Retrieve an access token for a current endpoint
+     */
+    String getEndpointAccessToken();
+
+    /**
+     * Updates with new endpoint attach request<br>
+     * <br>
+     * {@link org.kaaproject.kaa.client.event.registration.OnAttachEndpointOperationCallback} is populated with {@link org.kaaproject.kaa.client.event.EndpointKeyHash} of an
+     * attached endpoint.
+     *
+     * @param endpointAccessToken Access token of the attaching endpoint
+     * @param resultListener Listener to notify about result of the endpoint attaching
+     *
+     * @see org.kaaproject.kaa.client.event.EndpointAccessToken
+     * @see org.kaaproject.kaa.client.event.registration.OnAttachEndpointOperationCallback
+     */
+    void attachEndpoint(EndpointAccessToken endpointAccessToken, OnAttachEndpointOperationCallback resultListener);
+
+    /**
+     * Updates with new endpoint detach request
+     *
+     * @param endpointKeyHash Key hash of the detaching endpoint
+     * @param resultListener Listener to notify about result of the enpoint attaching
+     *
+     * @see org.kaaproject.kaa.client.event.EndpointKeyHash
+     * @see OnDetachEndpointOperationCallback
+     */
+    void detachEndpoint(EndpointKeyHash endpointKeyHash, OnDetachEndpointOperationCallback resultListener);
+
+    /**
+     * Creates user attach request using default verifier. Default verifier is selected during SDK generation.
+     * If there was no default verifier selected this method will throw runtime exception.
+     *
+     * @param userExternalId
```

Comment:
I think we have to add documentation for these parameters (both attachUser versions).